dashboard/src/actions/overviewActions.js

```diff
@@ -57,39 +54,36 @@ export const getDatasets = () => async (dispatch, getState) => {
         });
       }
     }
-    dispatch({ type: TYPES.COMPLETED });
   } catch (error) {
     dispatch(constructToast("danger", error?.response?.data?.message));
     dispatch({ type: TYPES.NETWORK_ERROR });
-    dispatch({ type: TYPES.COMPLETED });
   }
+  dispatch({ type: TYPES.COMPLETED });
 };
 
 export const updateDataset =
   (dataset, actionType, updateAction) => async (dispatch, getState) => {
     try {
-      dispatch({ type: TYPES.LOADING });
       const actions = {
         save: "dashboard.saved",
         read: "dashboard.seen",
         favorite: "user.favorite",
       };
+      dispatch({ type: TYPES.LOADING });
       const savedRuns = getState().overview.savedRuns;
       const newRuns = getState().overview.newRuns;
+      const initNewRuns = getState().overview.initNewRuns;
+
       const method = actions[actionType];
-      const metadata = { [method]: updateAction };
-      const params = {
-        name: dataset.name,
-        metadata,
-      };
+
       const dataIndex = newRuns.findIndex(
         (item) => item.resource_id === dataset.resource_id
       );
       const endpoints = getState().apiEndpoint.endpoints;
       const response = await API.put(
         `${endpoints?.api?.datasets_metadata}/${dataset.resource_id}`,
         {
-          ...params,
+          metadata: { [method]: updateAction },
```

Comment:
Yeah, I tripped over that syntax, too, and eventually chased down the answer:  in the context of an object literal, the key is a string (or a `Symbol`), either implicitly or explicitly, and so, since `method` is a _variable_, it needs to be dereferenced in order to obtain the string value (since we don't want the key to be `"method"`, here), and the JS syntax for that is to use square brackets.  So, this has nothing to do with arrays (it's more like a closure).

Yeah, `updateAction` isn't a good name (I suspect that it's supposed to be something more like "actionUpdate[Value]", and the "value" is silent.  😉  "updateValue" or "actionValue" would be better.